packages/vega-parser/src/parsers/guides/axis-util.js

```diff
@@ -0,0 +1,42 @@
+import { stringValue } from "vega-util";
+import { Top, Bottom, Left } from "./constants";
+
+export function xAxisExpr(signalExpr, ifXAxis, otherwise, ifXAxisIsString = true, otherwiseIsString = true) {
+  var ifXAxisStr = ifXAxisIsString ? stringValue(ifXAxis) : ifXAxis;
+  var otherwiseStr = otherwiseIsString ? stringValue(otherwise) : otherwise;
+  return {
+    signal: `${xAxisBooleanExpr(signalExpr)} ? (${ifXAxisStr}) : (${otherwiseStr})`
+  }
+}
+  
+export function xAxisBooleanExpr(signalExpr, isXAxis = true) {
```

Comment:
It also seems like `signalExpr` is not an arbitrary `signalExpr`, but rather `orientSignal`.  Better give a specific name.

Reply:
👍